src/vivarium/framework/results/manager.py

```diff
@@ -87,16 +87,20 @@ def on_post_setup(self, _: Event):
                         stratification.name: stratification.categories
                         for stratification in observation_stratifications
                     }
-                    # Get the complete cartesian product of stratifications
-                    index_values = list(itertools.product(*stratification_values.values()))
+                    if stratification_values:
+                        # Create index of the complete stratifications
+                        idx = pd.MultiIndex.from_tuples(
+                            list(itertools.product(*stratification_values.values())),
+                            names=stratification_values.keys(),
+                        )
+                    else:
+                        idx = pd.Index(aggregator_sources, name=measure)
```

Comment:
Rajan: hopefully this case goes away when we go back to using groupby objects (even for no stratifications)

Reply:
It doesn't work
```
(Pdb++) pd.MultiIndex.from_tuples([()], names=[])
*** ValueError: Length of names must match number of levels in MultiIndex.
Traceback (most recent call last):
  File "/ihme/homes/sbachmei/miniconda3/envs/dev-results-processing/lib/python3.11/site-packages/pandas/core/indexes/multi.py", line 222, in new_meth
    return meth(self_or_cls, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ihme/homes/sbachmei/miniconda3/envs/dev-results-processing/lib/python3.11/site-packages/pandas/core/indexes/multi.py", line 598, in from_tuples
    return cls(
           ^^^^
  File "/ihme/homes/sbachmei/miniconda3/envs/dev-results-processing/lib/python3.11/site-packages/pandas/core/indexes/multi.py", line 356, in __new__
    result._set_names(names)
  File "/ihme/homes/sbachmei/miniconda3/envs/dev-results-processing/lib/python3.11/site-packages/pandas/core/indexes/multi.py", line 1609, in _set_names
    raise ValueError(
```